components/inputs/input-search.js

```diff
@@ -0,0 +1,235 @@
+import '../button/button-icon.js';
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { classMap} from 'lit-html/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { inputStyles } from './input-styles.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class InputSearch extends LocalizeStaticMixin(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean },
+			label: { type: String },
+			lastSearchValue: { type: String, attribute: false },
+			maxlength: { type: Number },
+			noClear: { type: Boolean, attribute: 'no-clear' },
+			placeholder: { type: String },
+			value: { type: String },
+			_focussed: { type: Boolean, attribute: false },
+			_hovered: { type: Boolean, attribute: false }
+		};
+	}
+
+	static get styles() {
+		return [inputStyles, css`
+				:host {
+					display: inline-block;
+					position: relative;
+					width: 100%
+				}
+				:host([hidden]) {
```

Comment:
We may have talked about this and I missed it, but seems to be a pattern - are we adding this to all Lit components so the consumer doesn't need to worry about it?

Reply:
Any time we set a `display` value on the host, we need to specify this rule as the built-in browser rule will get overridden. It's not a Lit thing but more of a custom element thing.

We're following [Google's Best Practices](https://developers.google.com/web/fundamentals/web-components/best-practices#set-a-:host-display-style-e.g.-block-,-inline-block-,-flex--unless-you-prefer-the-default-of-inline-.) (there are 2 rules back to back there).